Makefile.in

```diff
@@ -73,11 +73,14 @@ OBJS = $(OBJC) $(OBJA)
 
 PIC_OBJS = $(PIC_OBJC) $(PIC_OBJA)
 
+# extra flags for crc-folding.c
+CRC_FOLDING_FLAGS = $(if $(findstring USE_VPCLMULQDQ_CRC,$(CFLAGS)), -mavx512f -mvpclmulqdq, )
```

Comment:
This make all code of crc-folding.c compiled with avx512f flag which may unwanted avx512 instruction code used for original piece of crc-folding.c. Better to split the vpclmulqdq implementation to new file with the required flag.

Reply:
For my understanding, once the code part is not executed, it will be not a problem.
Here we introduce a runtime flag (`x86_cpu_has_vpclmulqdq`) to check the capability.
I test the same binary on a Haswell machine, it can run properly without executing vpclmulqdq code.

Reply:
I think I got your point, you means the avx512 instruction may introduce by gcc to the original code, and make PCLMUL mode fail to run on non-AVX machine.
It' s a potential problem, I will try to fix it.

Reply:
Yes. It has potential risk here. We can't rely on compiler behavior.